src/encoder.rs

```diff
@@ -0,0 +1,115 @@
+use std::io;
+
+use crate::{AvifData, AvifImage};
+use libavif_sys as sys;
+
+pub struct Encoder {
+    encoder: *mut sys::avifEncoder,
+}
+
+impl Encoder {
+    pub fn new() -> Self {
+        let encoder = unsafe { sys::avifEncoderCreate() };
+        let mut encoder = Self { encoder };
+        encoder.set_speed(10);
+        encoder
+    }
+
+    pub fn max_threads(&self) -> usize {
+        unsafe { (*self.encoder).maxThreads as usize }
+    }
+
+    pub fn set_max_threads(&mut self, max_threads: usize) -> &mut Self {
+        assert!(max_threads >= 1, "max_threads must be >= 1");
+
+        unsafe { (*self.encoder).maxThreads = max_threads as i32 }
+        self
+    }
+
+    pub fn min_quantizer(&self) -> u8 {
+        unsafe { (*self.encoder).minQuantizer as u8 }
+    }
+
+    pub fn set_min_quantizer(&mut self, min_quantizer: u8) -> &mut Self {
+        assert!(min_quantizer <= 63, "min_quantizer must be <= 63");
+
+        unsafe { (*self.encoder).minQuantizer = min_quantizer as i32 }
+        self
+    }
+
+    pub fn max_quantizer(&self) -> u8 {
+        unsafe { (*self.encoder).maxQuantizer as u8 }
+    }
+
+    pub fn set_max_quantizer(&mut self, max_quantizer: u8) -> &mut Self {
+        assert!(max_quantizer <= 63, "max_quantizer must be <= 63");
+
+        unsafe { (*self.encoder).maxQuantizer = max_quantizer as i32 }
+        self
+    }
```

Comment:
Seeing one of your comments in squoosh.app would it make sense to just have a single quantizer value? @kornelski

Reply:
Yes, indeed